src/firstboot/firstboot.c

```diff
@@ -1040,6 +1042,69 @@ static int process_kernel_cmdline(int rfd) {
         return 0;
 }
 
+static int process_reset(int rfd) {
+        _cleanup_free_ char *esp_path = NULL, *random_seed = NULL;
+        _cleanup_close_ int esp_fd = -EBADF;
+        int r;
+
+        if (!arg_reset)
+                return 0;
+
+        FOREACH_STRING(s,
+                       "etc/locale.conf",
+                       "etc/vconsole.conf",
+                       "etc/localtime",
+                       "etc/hostname",
+                       "etc/machine-id",
+                       "etc/kernel/cmdline",
+                       "etc/machine-info",
+                       "var/lib/systemd/random-seed",
+                       "var/lib/systemd/credential.secret") {
+                _cleanup_free_ char *p = NULL;
+
+                r = chase_symlinks_at(rfd, s, CHASE_NOFOLLOW|CHASE_NONEXISTENT|CHASE_AT_RESOLVE_IN_ROOT, &p, NULL);
```

Comment:
here too, we should really pin the dirs the files are in, not the files. 

maybe we should add CHASE_PARENT as new flag that implicitly does path_extract_directory() on the passed path (but in a a root-dir-aware way), and then returns a pinned fd to the dir the specified file is in. The caller can then call path_extract_filename() on the original path and safely change the dentry relative to the pinned dir fd.